client.go

```diff
@@ -542,19 +545,26 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 		}
 	}
 
-	if c.ErrorHandler != nil {
+	defer func() {
```

Comment:
Pretty sure this is wrong as it will close all Idle Connections on every return, which means no keepalive will be used. In the non-error case below on lines 552, we keep them alive between successful requests.

Reply:
Wasn't my intent, let me look at this area again.

Reply:
Sorry that this took so long to wrap back around on. I'm going to be looking to address this, and the other feedback shortly.